src/modules/flow/max31855/max31855.c

```diff
@@ -0,0 +1,109 @@
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in
+ *     the documentation and/or other materials provided with the
+ *     distribution.
+ *   * Neither the name of Intel Corporation nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "max31855-gen.h"
+
+#include "sol-flow-internal.h"
+#include "sol-spi.h"
+#include "max31855-gen.h"
+
+#include <sol-util.h>
+#include <errno.h>
+
+struct max31855_options {
+    struct sol_spi *device;
+    unsigned int bus;
+    unsigned int chip_select;
+};
+
+static int
+max31855_open(struct sol_flow_node *node, void *data, const struct sol_flow_node_options *options)
+{
+    struct max31855_options *mdata = data;
+    const struct sol_flow_node_type_aio_max31855_options *opts;
+
+    SOL_FLOW_NODE_OPTIONS_SUB_API_CHECK(options,
+        SOL_FLOW_NODE_TYPE_AIO_MAX31855_OPTIONS_API_VERSION,
+        -EINVAL);
+    opts = (const struct sol_flow_node_type_aio_max31855_options *)options;
+
+    mdata->bus = opts->bus.val;
+    mdata->chip_select = opts->chip_select.val;
+    mdata->device = sol_spi_open(mdata->bus, mdata->chip_select);
+
+    if (!mdata->device)
+        return -ENOMEM;
+
+    sol_spi_set_max_speed(mdata->device, 2000000);
+
+    return 0;
+}
+
+static void
+max31855_close(struct sol_flow_node *node, void *data)
+{
+    struct max31855_options *mdata = data;
+
+    sol_spi_close(mdata->device);
+}
+
+static int
+max31855_process(struct sol_flow_node *node, void *data, uint16_t port, uint16_t conn_id, const struct sol_flow_packet *packet)
+{
+    struct max31855_options *mdata = data;
+    int32_t raw_value;
+    double value;
+    uint8_t buf[4];
+
+    memset(buf, 0, sizeof(buf));
+
+    if (!sol_spi_raw_transfer(mdata->device, buf, sizeof(buf)))
```

Comment:
it was a direct translation of the upm code to something that we had on sol_spi:
 uint8_t* x = mraa_spi_write_buf(m_sensor, buf, 4);
it seemed like the sol_spi_raw_transfer should be the right one, since I'm not sending a tx to receive a rx, I just read what's in the device to the buffer.